src/Features/Core/Portable/MetadataAsSource/MetadataAsSourceFileService.cs

```diff
@@ -165,68 +129,12 @@ private static void AssertIsMainThread(MetadataAsSourceWorkspace workspace)
         Contract.ThrowIfFalse(threadingService.IsOnMainThread);
     }
 
-    public bool TryAddDocumentToWorkspace(string filePath, SourceTextContainer sourceTextContainer, [NotNullWhen(true)] out DocumentId? documentId)
```

Comment:
these are now implemented by the host (VS workspace).  The host decides if opening a document in the ws is important